### PR TITLE
(cheevos) delay subsequent unlock attempts on failure

### PR DESCRIPTION
**Pull request by Jamiras**
## Description

Adds a `when` property to `retro_task_t`. When processing the queue in threaded mode, the thread will sleep until the next scheduled task via `scond_wait_timeout`. In non-threaded mode, tasks that shouldn't run yet are ignored. If `when` is left at 0, the task will run as soon as possible.

This fuctionality has been hooked up to the achievement unlock retry code. The first retry will be scheduled with a 250ms delay. Each additional retry will double the wait until the retries are spaced two minutes apart.

When investigating #10138, I noticed that the retry loop was very tight. As soon as the failure occurred, the request would be immediately requeued, and this would continue over and over again until the request finally went through. It's a very busy loop, as the network failure would return an error almost immediately. 

The trade-off to this implementation is that once network connectivity is restored, it may still take up to two minutes before the unlocks are sent. It seems unlikely that users are accustomed to playing "offline" and reconnecting for a brief moment to flush their unlocks before shutting down the application, but if they do, they could be impacted.

Additionally, the rich presence functionality has been moved to a two-minute scheduled task. This ensures the rich presence information is sent to the server even if the user is in the RetroArch menu or another application, which ensures they remain on the Active Users list on retroachievements.org.

A similar change should be made for leaderboard submits as well, but I wanted approval on the approach/implementation first.

## Related Issues

#10138 - this addresses the busy loop. I do not claim that it fixes the original problem, which I have not been able to reproduce.

## Related Pull Requests

n/a

## Reviewers

@twinaphex

### CODE REVIEW

**Comment by twinaphex** [comment]
OK, so when 'when' is 0, there should be no functional change from before, right?

**Comment by Jamiras** [comment]
Correct. In both the threaded and non-threaded models, if `when` is 0, the flow is exactly the same, and since `task_init` uses `calloc`, it should be 0 for all existing tasks with no additional changes.

File: libretro-common/queues/task_queue.c [hunk 1]
```diff
@@ -320,6 +340,14 @@ static void task_queue_remove(task_queue_t *queue, retro_task_t *task)
       {
          t->next    = task->next;
          task->next = NULL;
+
```
Comment (twinaphex) [inline libretro-common/queues/task_queue.c]: Can this part be documented a bit with a comment?

Comment (twinaphex) [reply libretro-common/queues/task_queue.c]: After this conditional is documented, I think we can go ahead and merge this.

**Pull request merged**